src/v/cluster/partition_balancer_planner.cc

```diff
@@ -67,12 +67,6 @@ void partition_balancer_planner::init_per_node_state(
   reallocation_request_state& rrs,
   plan_data& result) const {
     for (const auto& [id, broker] : _state.members().nodes()) {
-        if (
-          broker.state.get_membership_state()
-          == model::membership_state::removed) {
```

Comment:
since `removed` is deprecated, would it make sense to rename it (so that it's clear it should never be used except maybe in decoding a message where it is discarded) or having an assertion or something that it is never used? i'm really not sure, maybe it's overkill.